modules/misc/xdg-user-dirs.nix

```
@@ -86,25 +86,41 @@ in {
       example = { XDG_MISC_DIR = "$HOME/Misc"; };
       description = "Other user directories.";
     };
+
+    createDirs = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to automatically create the XDG user directories.
+      '';
+    };
   };
 
-  config = mkIf cfg.enable {
+  config = let
+    directories = {
+      XDG_DESKTOP_DIR = cfg.desktop;
+      XDG_DOCUMENTS_DIR = cfg.documents;
+      XDG_DOWNLOAD_DIR = cfg.download;
+      XDG_MUSIC_DIR = cfg.music;
+      XDG_PICTURES_DIR = cfg.pictures;
+      XDG_PUBLICSHARE_DIR = cfg.publicShare;
+      XDG_TEMPLATES_DIR = cfg.templates;
+      XDG_VIDEOS_DIR = cfg.videos;
+    } // cfg.extraConfig;
+  in mkIf cfg.enable {
     xdg.configFile."user-dirs.dirs".text = let
-      options = {
-        XDG_DESKTOP_DIR = cfg.desktop;
-        XDG_DOCUMENTS_DIR = cfg.documents;
-        XDG_DOWNLOAD_DIR = cfg.download;
-        XDG_MUSIC_DIR = cfg.music;
-        XDG_PICTURES_DIR = cfg.pictures;
-        XDG_PUBLICSHARE_DIR = cfg.publicShare;
-        XDG_TEMPLATES_DIR = cfg.templates;
-        XDG_VIDEOS_DIR = cfg.videos;
-      } // cfg.extraConfig;
-
       # For some reason, these need to be wrapped with quotes to be valid.
-      wrapped = mapAttrs (_: value: ''"${value}"'') options;
+      wrapped = mapAttrs (_: value: ''"${value}"'') directories;
     in generators.toKeyValue { } wrapped;
 
     xdg.configFile."user-dirs.conf".text = "enabled=False";
+
+    home.activation = mkIf cfg.createDirs {
+      createXdgUserDirectories = let
+        directoriesList = attrsets.attrValues directories;
```

I think it should work with just
```suggestion
        directoriesList = attrValues directories;
```

---

Seems so:

```
nix-repl> lib.attrValues
«primop»
```